### DISCUSSION TITLE
Write SilentTimer bei Wärmepumpe Grundsatzfrage

### DISCUSSION THREAD

**Discussion by xerion3800** [john30/ebusd#1125] [Q&A]
Hallo zusammen,
ich versuche den silentTimer zu setzen aber habe offensichtlich ein Problem mit der Formatierung der Daten und weiß  leider nicht wie ich herausfinde welchen Wert ich übergeben muss. Hat da jemand einen Tipp für mich?
Ich versuche also mit meiner15.ctlv2.csv einer der folgenden Timer zu setzen:

```
# timers silent mode,,,,,,,,,,,,,
*r,,,,,,B555,a50004,,,,,,
*w,,,,,,B555,a60004,,,,,,
r,,silentTimer.Config,timer Configuration(static atm? describes featue set?),,,B555,a30004,,,HEX:9,,,Configuration
r,,silentTimer.Timeframes,Configured slot count,,,B555,a40004,,,slotCountWeek,,,Amount of configured slots per weekday
r,,silentTimer.Monday0,timer monday 1,,,,0000,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Monday1,timer monday 2,,,,0001,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Monday2,timer monday 3,,,,0002,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday0,timer tuesday 1,,,,0100,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday1,timer tuesday 2,,,,0101,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday2,timer tuesday 3,,,,0102,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday0,timer wednesday 1,,,,0200,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday1,timer wednesday 2,,,,0201,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday2,timer wednesday 3,,,,0202,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday0,timer thursday 1,,,,0300,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday1,timer thursday 2,,,,0301,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday2,timer thursday 3,,,,0302,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday0,timer friday 1,,,,0400,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday1,timer friday 2,,,,0401,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday2,timer friday 3,,,,0402,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday0,timer saturday 1,,,,0500,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday1,timer saturday 2,,,,0501,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday2,timer saturday 3,,,,0502,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday0,timer sunday 1,,,,0600,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday1,timer sunday 2,,,,0601,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday2,timer sunday 3,,,,0602,,,rTimeSlotWithoutTemp,,,
w,,silentTimer.Monday,timer monday,,,,00,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Tuesday,timer tuesday,,,,01,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Wednesday,timer wednesday,,,,02,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Thursday,timer thursday,,,,03,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Friday,timer friday,,,,04,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Saturday,timer saturday,,,,05,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Sunday,timer sunday,,,,06,,,wTimeSlotWithoutTemp,,,
```
Wenn ich z.B. folgenden Aufruf nutze bekomme ich folgende Rückgabe:
```
fort@FHEM:~/ebus/ebusd-configuration$ ebusctl write -c ctlv2 silentTimer.Monday '00:00;09:00'   
ERR: invalid numeric argument
```
Wenn ich z.B. `silentTimer.Monday0` abfrage bekomme ich das zurück:
```
fort@FHEM:~/ebus/ebusd-configuration$ ebusctl read -f silentTimer.Monday0   
00:00;00:00
```

**Comment by janpedder** [comment]
Moin,
Das Elend zu Erst: ich hab keine Ahnung.

Beim Schreiben auf den Bus bin ich noch nicht angekommen, aber im folgenden wurde das Schreiben der anderen Zeitprogramme thematisiert. Vielleicht hilft dir das. Es gibt bestimmt Zielführenderes als den verlinkten Thread, da auch noch ein timeslotwithtemp geschrieben wird, aber das wollte ich nicht vorenthalten:
[https://github.com/john30/ebusd-configuration/pull/266#issuecomment-1345345618](https://github.com/john30/ebusd-configuration/pull/266#issuecomment-1345345618)

Woher kommen hier die Definitionen für den silenttimer? Hab die spasseshalber in meine 15.ctlv2.csv eingefügt und es kommen plausible Daten an. Im hier hinterlegten 15.ctlv2.csv Template sind die nicht drin, oder verballer ich irgendwas?

**Reply by xerion3800** [reply]
Erst mal danke für den Ansatz das schaue ich mir an. 
Woher die Daten kommen weiß ich ehrlich gesagt nicht mehr. Habe mit in letzter Zeit über verschiedene Quellen die Sachen zusammen gesucht und abgespeichert sofern dort Daten plausibel angefragt werden konnten.

**Reply by janpedder** [reply]
Wärst du bereit, deine csv hier einzustellen?
Vielleicht hast du noch ein paar mehr Datenpunkte, die hier fehlen.

**Reply by xerion3800** [reply]
Ich habe das Repo von John geforked und meine Sachen ergänzt. Kannst gerne mal drüber schauen.
https://github.com/xerion3800/ebusd-configuration/tree/master/ebusd-2.1.x/en/vaillant

**Reply by janpedder** [reply]
moin, hab den silenttimer schreiben können.

ich habe aus bereits o.g. thread [https://github.com/john30/ebusd-configuration/pull/266#issuecomment-1473275155](https://github.com/john30/ebusd-configuration/pull/266#issuecomment-1473275155) den workaround genutzt.
Die dazugehörige ebusd-2.1.x/de/vaillant/_templates.csv aus [janphkre:feature/support-ctlv2](https://github.com/janphkre/ebusd-configuration/tree/feature/support-ctlv2) bei mir eingesetzt.
entscheidend hier:
```
wTimeSlotWithoutTemp,slotIndex;slotCount;slotTimeFrame;HEX:2,,,Index des Slot; gesamt Anzahl der aktiven Slots und einzelner Zeitslot ohne Temperatur;FF FF muss angehangen werden
```

so funktioniert dann mit den Definitionen von dir :
```
ebusctl write -c ctlv2 silenttimer.Friday "0;3;00:20;02:20;ff ff";
ebusctl write -c ctlv2 silenttimer.Friday "1;3;06:30;08:30;ff ff";
ebusctl write -c ctlv2 silenttimer.Friday "2;3;10:40;23:40;ff ff"
```
```
ebusctl read -c ctlv2 silenttimer.Friday0;
ebusctl read -c ctlv2 silenttimer.Friday1;
ebusctl read -c ctlv2 silenttimer.Friday2
```

dauert je nach Abfrageintervall, bis die geschriebenen Daten aktualisiert sind, im Regler selbst können die aber sofort abgelesen werden.
Werte können nur in 10 Minuten Intervallen geschrieben werden.

_______________________________________________________________________________________________________________


edit:
Zusammengefasst:
diese config: [https://github.com/janphkre/ebusd-configuration](https://github.com/janphkre/ebusd-configuration)
daraus die 15.ctlv2.csv ergänzt mit:
```
# timers silent mode,,,,,,,,,,,,,
timers silent mode,,,,,,,,,,,,,
*r,,,,,,B555,a50004,,,,,,
*w,,,,,,B555,a60004,,,,,,
r,,silentTimer.Config,Zeitfenster Konfiguration(aktuell statisch? Beschreibt eine Art Funktionalitaetsumfang)),,,B555,a30004,,,HEX:9,,,Konfiguration 
r,,silentTimer.Timeframes,Zeitfenster Anzahl,,,B555,a40004,,,slotCountWeek,,,Anzahl der eingestellten Slots pro Wochentag
r,,silentTimer.Monday0,timer monday 1,,,,0000,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Monday1,timer monday 2,,,,0001,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Monday2,timer monday 3,,,,0002,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday0,timer tuesday 1,,,,0100,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday1,timer tuesday 2,,,,0101,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Tuesday2,timer tuesday 3,,,,0102,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday0,timer wednesday 1,,,,0200,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday1,timer wednesday 2,,,,0201,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Wednesday2,timer wednesday 3,,,,0202,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday0,timer thursday 1,,,,0300,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday1,timer thursday 2,,,,0301,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Thursday2,timer thursday 3,,,,0302,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday0,timer friday 1,,,,0400,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday1,timer friday 2,,,,0401,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Friday2,timer friday 3,,,,0402,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday0,timer saturday 1,,,,0500,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday1,timer saturday 2,,,,0501,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Saturday2,timer saturday 3,,,,0502,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday0,timer sunday 1,,,,0600,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday1,timer sunday 2,,,,0601,,,rTimeSlotWithoutTemp,,,
r,,silentTimer.Sunday2,timer sunday 3,,,,0602,,,rTimeSlotWithoutTemp,,,
w,,silentTimer.Monday,timer monday,,,,00,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Tuesday,timer tuesday,,,,01,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Wednesday,timer wednesday,,,,02,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Thursday,timer thursday,,,,03,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Friday,timer friday,,,,04,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Saturday,timer saturday,,,,05,,,wTimeSlotWithoutTemp,,,
w,,silentTimer.Sunday,timer sunday,,,,06,,,wTimeSlotWithoutTemp,,,

```

**Reply by jakolion** [reply]
Aus irgendeinem Grund bin ich nicht in der Lage die Zeiten zu ändern. Wenn ich beispielsweise die Zeiten für Warmwasser versuche zu ändern, bekomme ich mit "done" eine positive Rückmeldung. Nach weit mehr als 10 Minuten haben sich die Werte aber nicht aktualisiert.

`ebusctl write -c 720 hwcTimer.Friday "0;3;10:00;20:30;50;ff ff"
done`

EDIT: Ich konnte es eben selbst lösen: ich dachte der zweite Wert sind die maximalen Slots, aber es beschreibt die aktuell genutzten Slots. Da ich nur einen nutze, musste ich die "3" durch eine "1" ersetzen und schon ging es.
